src/bucket.ts

```diff
@@ -1301,8 +1301,12 @@ class Bucket extends ServiceObject {
     );
 
     if (options.append === false) {
-      this.setMetadata({lifecycle: {rule: newLifecycleRules}}, callback);
-      this.storage.retryOptions.autoRetry = this.instanceRetryValue;
+      this.setMetadata({lifecycle: {rule: newLifecycleRules}})
+        .then(resp => callback!(null, resp))
```

Comment:
will callbacks always be provided? (guessing yes, but just wanted to double check this)